src/main/kotlin/org/malachite/estella/task/api/TaskController.kt

```diff
@@ -3,107 +3,129 @@ package org.malachite.estella.task.api
-            @RequestParam("taskStage", required = false) taskStageUuid: String?,
-            @RequestHeader(EStellaHeaders.devPassword) password: String
+        @RequestParam("owner", required = false) organizationUuid: String?,
+        @RequestParam("taskStage", required = false) taskStageUuid: String?,
+        @RequestHeader(EStellaHeaders.devPassword) password: String
     ): ResponseEntity<Any> {
-        if ((organizationUuid == null && taskStageUuid == null) || (organizationUuid != null && taskStageUuid != null))
-            return ResponseEntity.badRequest().body(Message("Exactly one of parameters: organizationUuid and taskStageUuid is required"))
+        if (listOfNotNull(organizationUuid, taskStageUuid).size != 1)
+            return ResponseEntity.badRequest().body(bothUUIDErrorMessage)
         val tasks: List<TaskDto> = organizationUuid
-                ?.let {
-                    taskService.checkDevPassword(it, password)
-                            .getTasksByOrganizationUuid(it)
-                }
-                ?: taskStageUuid
-                        ?.let { taskService.getTasksByTasksStage(it, password) }!!
+            ?.let {
+                taskService.checkDevPassword(it, password)
+                    .getTasksByOrganizationUuid(it)
+            }
+            ?: taskStageUuid?.let { taskService.getTasksByTasksStage(it, password) }!!
        return ResponseEntity.ok(tasks)
     }
 
-
-    @Deprecated(message = "Wasn't tested yet - unnecessary now - to be implemented and tested in ES-17 epic")
     @CrossOrigin
-    @Transactional
     @GetMapping("/{taskId}")
     fun getTaskById(
-            @RequestParam("owner") organizationUuid: String,
-            @PathVariable taskId: Int,
-            @RequestHeader(EStellaHeaders.devPassword) password: String
-    ) =
-            ResponseEntity(Message("Not Implemented"), HttpStatus.NOT_IMPLEMENTED) /*taskService.getTaskById(taskId)
-        .let { ResponseEntity.ok(it) }*/
+        @RequestParam("owner") organizationUuid: String,
+        @PathVariable taskId: Int,
+        @RequestHeader(EStellaHeaders.devPassword) password: String
+    ) = taskService
+        .checkDevPassword(organizationUuid, password)
+        .checkOrganizationRights(organizationUuid, taskId)
+        .getTaskById(taskId)
+        .let { ResponseEntity.ok(it) }
 
-    @Deprecated(message = "Wasn't tested yet - unnecessary now - to be implemented and tested in ES-17 epic")
     @CrossOrigin
-    @Transactional
     @GetMapping("/{taskId}/tests")
     fun getTaskTests(
-            @RequestParam("owner") organizationUuid: PayloadUUID,
-            @PathVariable taskId: Int,
-            @RequestHeader(EStellaHeaders.devPassword) password: String
-    ) =
-            ResponseEntity(Message("Not Implemented"), HttpStatus.NOT_IMPLEMENTED)
-    /*taskService.getTestsFromTask(taskId)
-        .let { ResponseEntity.ok(it) } */
+        @RequestParam("owner") organizationUuid: String,
```

Comment:
Ogólnie nie wiem czy ta zmiana ma sens bo wydaje mi się że powinniśmy jak najwcześniej zacząć operować na UUID lub czymś co w miarę szybko może się stać UUID, a nie potem w serwisach to mapować

Reply:
Bump